Implement toList(), deprecate toArray() (#186)
sanmai/pipeline: README.md

| `each()`     | Eagerly iterates over the sequence. | `foreach`, `array_walk` |
| `stream()` | Ensures subsequent operations use lazy, non-array paths | |
| `runningCount()` | Counts seen values using a reference argument. | |
| `toList()` | Returns an array with all values. Eagerly executed. |  |

> | `toAssoc()` | Returns an array with all values and keys. Eagerly executed. | `dict`, `ToDictionary` |

Author: ```suggestion
| `toAssoc()` | Returns a final array with values and keys. Eagerly executed. | `dict`, `ToDictionary` |
```
